### PR TITLE
[ARCHETYPE-531] Treat no modules element as no modules at all

### PR DESCRIPTION
**Pull request by hadrabap**
Treat no `<modules />` element as no modules at all

### CODE REVIEW

File: archetype-common/src/main/java/org/apache/maven/archetype/creator/FilesetArchetypeCreator.java [hunk 1]
```diff
@@ -702,7 +702,9 @@ private void rewriteEARPluginReferences( Plugin plugin, String rootArtifactId, S
         Xpp3Dom configuration = (Xpp3Dom) plugin.getConfiguration();
         if ( configuration != null )
         {
-            Xpp3Dom[] modules = configuration.getChild( "modules" ).getChildren();
+            Xpp3Dom[] modules = configuration.getChild( "modules" ) != null
```
Comment (Tibor17) [inline archetype-common/src/main/java/org/apache/maven/archetype/creator/FilesetArchetypeCreator.java:705]: use a local variable for `configuration.getChild( "modules" )`.
That's better than this and better for debugging as well.

Comment (hadrabap) [reply archetype-common/src/main/java/org/apache/maven/archetype/creator/FilesetArchetypeCreator.java]: Hi, I've incorporated your suggestion.

File: maven-archetype-plugin/src/it/projects/ARCHETYPE-531_ear-modules/pom.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (elharo) [inline maven-archetype-plugin/src/it/projects/ARCHETYPE-531_ear-modules/pom.xml:20]: we tend toward two space indents for XML files

**Comment by elharo** [comment]
Jenkins passed: https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-archetype/job/27/

**Comment by jira-importer** [comment]
Resolve #794

**Comment by jira-importer** [comment]
Resolve #794

**Pull request merged**